packages/databricks-vscode/src/configuration/ConnectionManager.ts

```diff
@@ -98,125 +153,53 @@ export class ConnectionManager {
         return this._workspaceClient?.apiClient;
     }
 
-    async login(interactive = false, force = false): Promise<void> {
-        try {
-            await this._login(interactive, force);
-        } catch (e) {
-            NamedLogger.getOrCreate("Extension").error("Login Error:", e);
-            if (interactive && e instanceof Error) {
-                window.showErrorMessage(`Login error: ${e.message}`);
-            }
-            this.updateState("DISCONNECTED");
-            await this.logout();
-        }
-    }
-
-    private async _login(interactive = false, force = false): Promise<void> {
+    private async login(
+        authProvider: AuthProvider,
+        force = false
+    ): Promise<void> {
         if (force) {
             await this.logout();
         }
         if (this.state !== "DISCONNECTED") {
             return;
         }
-        this.updateState("CONNECTING");
-
-        let projectConfigFile: ProjectConfigFile;
-        let workspaceClient: WorkspaceClient;
-
-        try {
-            try {
-                projectConfigFile = await ProjectConfigFile.load(
-                    vscodeWorkspace.rootPath!,
-                    this.cli.cliPath
-                );
-            } catch (e) {
-                if (
-                    e instanceof ConfigFileError &&
-                    e.message.startsWith("Project config file does not exist")
-                ) {
-                    this.updateState("DISCONNECTED");
-                    await this.logout();
-                    return;
-                } else {
-                    throw e;
-                }
-            }
-
-            if (!(await projectConfigFile.authProvider.check())) {
-                throw new Error(
-                    `Can't login with ${projectConfigFile.authProvider.describe()}.`
-                );
-            }
-
-            workspaceClient =
-                projectConfigFile.authProvider.getWorkspaceClient();
-
-            await workspaceClient.config.authenticate(new Headers());
-
-            this._databricksWorkspace = await DatabricksWorkspace.load(
-                workspaceClient,
-                projectConfigFile.authProvider
-            );
-        } catch (e: any) {
-            const message = `Can't login to Databricks: ${e.message}`;
-            NamedLogger.getOrCreate("Extension").error(message, e);
-            window.showErrorMessage(message);
 
-            this.updateState("DISCONNECTED");
-            await this.logout();
+        if (!(await authProvider.check())) {
+            // We return without any state changes. This ensures that
+            // if users move from a working auth type to an invalid
+            // auth, the old auth will continue working and they will not
+            // have to re authenticate even the old one.
```

Comment:
Will they get a notification about incorrect auth? 

I might be misunderstanding this comment, but if I change auth params to some incorrect values, I would expect the extension to stop working. Or at least notify me about the problem.

Should we throw an error here instead of silently returning, so the config wizard can show an error notification? (`init` login can also be wrapped)

Reply:
The errors and retries are handled by each of the "check" functions. https://github.com/databricks/databricks-vscode/blob/bundle-integ/packages/databricks-vscode/src/configuration/auth/AuthProvider.ts#L53-L57